### ISSUE TITLE
Initialize and manage Postgres and Mysql schema with Flyway

### ISSUE THREAD

**Issue by nterry** [kubedb/project#670]
I propose adding flyway support for initializing a Postgres or Mysql database. [Flyway](https://flywaydb.org) is a widely used technology and the scripts, while using sql syntax, don't have commit statements in them as they rely on flyway to handle state.

Specifically, flyway uses a naming pattern with each sql file representing an atomic, versioned change to the schema. See [flyway migrations](https://flywaydb.org/documentation/migrations) for more details.

I am proposing the following addition to the CRD:

    apiVersion: kubedb.com/v1alpha1
    kind: <Postgres|MySQL>
    ...
    schemaManagement:
      flyway:
        configMaps:
          namePrefix: pg-flyway-scripts

The namePrefix is to specify the _set_ of config maps that are prefixed with the given string as flyway reads a set of files, each representing a revision to the schema.

I propose that this will be run every single time the operator schedules an update to a managed database allowing the schema to be revved any time a new config map with the correct namePrefix is added. Any errors will be left up to the developer to address.

Thoughts?

**Comment by chris13524** [comment]
I agree, we need a good cloud-native solution for database migrations. But I think the migration scripts should be derived from version control or a Docker image, not a volume. And using a config map for storing migrations I feel would be inefficient to store in the Kube API.

Some related things I found:

 - https://cloud-native-schema-migrations.github.io/
 - https://github.com/nelvadas/ocp-flyway-db-migration
 - https://cloudnativedevopsblog.com/posts/database-migrations/

Seems the current approach is using a Job that points to a Docker image (that you update) that runs the latest migrations. Can be combined with Flux or your favourite CI tool to update the image and the Job.
